h2o-core/src/main/java/hex/Model.java

```diff
@@ -982,10 +980,16 @@ else if (isWeights && computeMetrics) {
             msgs.add(H2O.technote(1, "Test/Validation dataset is missing weights column '" + names[i] + "' (needed because a response was found and metrics are to be computed): substituting in a column of 1s"));
           }
         } else if (expensive) {   // generate warning even for response columns.  Other tests depended on this.
```

Comment:
Please look at the lines 978-983 of the original code. If weights column should be included and we don't want to calculate metrics the weights column will be filled with NAs instead of 1 (which would be the case if we wanted metrics). This doesn't make logical sense.

If weights column was not provided by the user it should have the same default value regardless if the user provided response or not (= the user wants to calculate metrics or not).